File: algorithms/1_easy/C/01502_can_make_arithmetic_progression_from_sequence/solution.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        const int diff{arr[1] - arr[0]};

        for (auto it{arr.cbegin() + 2}; it != arr.cend(); ++it) {
            if (*it - *(it - 1) != diff) {
                return false;
            }
        }

        return true;
    }
};

File: algorithms/1_easy/C/01502_can_make_arithmetic_progression_from_sequence/solution.rs
impl Solution {
    pub fn can_make_arithmetic_progression(mut arr: Vec<i32>) -> bool {
        arr.sort_unstable();
        let diff = arr[1] - arr[0];

        for window in arr.windows(2).skip(1) {
            if window[1] - window[0] != diff {
                return false
            }
        }

        true
    }
}
